dds/idl/typeobject_generator.cpp

```diff
@@ -1762,7 +1762,11 @@ typeobject_generator::generate(AST_Type* node, UTL_ScopedName* name)
     gti.endArgs();
     const OpenDDS::XTypes::TypeIdentifier ti = get_minimal_type_identifier(node);
     be_global->impl_ <<
-      "  static const XTypes::TypeIdentifier ti = " << ti << ";\n"
+      "  ACE_GUARD_RETURN(ACE_Thread_Mutex, guard, TheServiceParticipant->get_tm_lock(), ti);\n"
+      "  static XTypes::TypeIdentifier ti;\n"
```

Comment:
Now this looks like a problem because (in the generated code) we have ti referenced before it's defined.